### PR TITLE
[SPARK-7957] Preserve partitioning when using randomSplit

### PR DESCRIPTION
**Pull request by brkyvz**
cc @JoshRosen 
Thanks for noticing this!

### CODE REVIEW

**Comment by JoshRosen** [comment]
LGTM pending Jenkins.

File: core/src/main/scala/org/apache/spark/rdd/RDD.scala [hunk 1]
```diff
@@ -434,11 +434,11 @@ abstract class RDD[T: ClassTag](
    * @return A random sub-sample of the RDD without replacement.
    */
   private[spark] def randomSampleWithRange(lb: Double, ub: Double, seed: Long): RDD[T] = {
-    this.mapPartitionsWithIndex { case (index, partition) =>
+    this.mapPartitionsWithIndex( { case (index, partition) =>
       val sampler = new BernoulliCellSampler[T](lb, ub)
       sampler.setSeed(seed + index)
       sampler.sample(partition)
-    }
+    }, true)
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/rdd/RDD.scala]: can you used name argument for this?

File: core/src/main/scala/org/apache/spark/rdd/RDD.scala [hunk 2]
```diff
@@ -434,11 +434,11 @@ abstract class RDD[T: ClassTag](
    * @return A random sub-sample of the RDD without replacement.
    */
   private[spark] def randomSampleWithRange(lb: Double, ub: Double, seed: Long): RDD[T] = {
-    this.mapPartitionsWithIndex { case (index, partition) =>
+    this.mapPartitionsWithIndex( { case (index, partition) =>
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/rdd/RDD.scala]: nitpick - you don't actually need the case here

File: core/src/main/scala/org/apache/spark/rdd/RDD.scala (hunk 1 above)
Comment (JoshRosen) [inline core/src/main/scala/org/apache/spark/rdd/RDD.scala]: I thought about nitpicking on this and second-guessed myself, but I actually agree; let's name this parameter.

**Comment by rxin** [comment]
Merged. Thanks.

**Pull request closed without merging**